### PR TITLE
Expose client IP address to server

### PR DESCRIPTION
**Pull request by bgetsug**
I probably should have opened an issue first, but I wanted to be able to log the client IP address in an `Interceptor`. Unless I missed this already being available somewhere, here's my solution. Thoughts?

### CODE REVIEW

File: lib/src/server/call.dart [hunk 1]
```diff
@@ -44,6 +44,9 @@ abstract class ServiceCall {
   /// Returns the client certificate if it is requested and available
   X509Certificate? get clientCertificate;
 
+  /// Returns the IP address of the client, if available
+  InternetAddress? get clientIpAddress;
```
Comment (mraleph) [inline lib/src/server/call.dart:48]: Could you rename this into `remoteAddress` to match `dart:io` `Socket` API naming?

**Review by mraleph — CHANGES_REQUESTED** [review]
LGTM with a minor nit. 

(Sorry for the delay - I was away on vacation).

File: lib/src/server/call.dart (hunk 1 above)
Comment (bgetsug) [reply lib/src/server/call.dart]: ✅ 😄

**Comment by mraleph** [comment]
LGTM! Thanks.

File: lib/src/server/server.dart [hunk 2]
```diff
@@ -107,16 +107,21 @@ class ConnectionServer {
 
   Service? lookupService(String service) => _services[service];
 
-  Future<void> serveConnection(
-    ServerTransportConnection connection, [
+  Future<void> serveConnection({
```
Comment (natebosch) [inline lib/src/server/server.dart:110]: This was a breaking change.

File: lib/src/server/server.dart [hunk 3]
```diff
@@ -228,13 +236,23 @@ class Server extends ConnectionServer {
...
+
+      final connection = ServerTransportConnection.viaSocket(
+        socket,
+        settings: http2ServerSettings,
+      );
+
+      serveConnection(
+        connection: connection,
+        clientCertificate: clientCertificate,
+        remoteAddress: socket.remoteAddress,
```
Comment (whesse) [inline lib/src/server/server.dart:254]: This line, fetching socket.remoteAddress, has created an uncaught exception in the case where the socket is already closed when we reach this point.

Previously, the server worked if a server connection was requested for a closed socket, and did not throw an exception.  I had a server running for 16 months without failing using the previous code, and running it with this code failed in less than 8 hours.

**Pull request merged**